TV Show Project | Delroy Gayle (DelroyGayle)

DelroyGayle opened the pull request:
Levels 100 to 300

DelroyGayle committed:
Update index.html

Added my name

DelroyGayle committed:
Levels 100 to 300 done

RitaGlushkova commented on script.js:
```diff
@@ -1,12 +1,273 @@
+
+// Instead of using innerHtml, remove the <p></p> and use innerText instead
+function handle_paragraph(text) {
+      text = text.replace(/<\/?p>/ig,""); // remove any <p> </p>
```
There is also a couple of paragraphs, where you need to remove <br>. Check Hardhome - S05E08 and The Gift - S05E07

DelroyGayle replied:
Thank you

RitaGlushkova commented on script.js:
```diff
@@ -1,12 +1,273 @@
+      let len = lowerCase.length;
+
+      // Add a <p></p> to create a space underneath the image
+      paragraph = document.createElement("p");
```
You can also use CSS class for padding and ad it to your image div or summary.

RitaGlushkova commented on script.js:
```diff
@@ -1,12 +1,273 @@
+            lowerText = lowerText.substr(p+len);
+            paragraph = document.createElement("span");
+            paragraph.innerText=aString;
+            paragraph.classList.add('blueText');
```
Nice solution for highlighting matching string!!!

RitaGlushkova commented on script.js:
```diff
@@ -1,12 +1,273 @@
+    // in order to highlight
+
+    text = source.summary.replace(/<\/?p>/ig,""); // remove any <p> </p>
+    processText(text,lowerCase,"",episodeDiv);
```
Is it repeating lines 120-122?

DelroyGayle replied:
No! One is the episode followed by S01E01 etc so Full
The other is simply S01E01 so Suffix

RitaGlushkova commented on script.js:
```diff
@@ -1,12 +1,273 @@
+    // add to main content
+    mainDisplayDiv.appendChild(episodeDiv); 
+
+    if ((index+1) % 3 === 0) // MAKE THE 4TH COLUMN A GAP BY CREATING AN EMPTY DIV
```
You already have a class .gridDisplay, which has padding-left: 3em
As an option, you could do the same for padding-right instead of an empty <did>

DelroyGayle replied:
Thank you for the other solution that you showed me

RitaGlushkova commented on script.js:
```diff
@@ -1,12 +1,273 @@
+  option.value = "";
+  option.text = "Select an episode ..."; // Placeholder
+  selectMenu.appendChild(option);
+  showAll(true); // display all episodes
```
Excellent solution! And I like that it refreshes as well!

RitaGlushkova reviewed (commented):
Great work Delroy! Lots of good functionality was added to the website. I am especially impressed with your solution for the default select menu. 

I left a few comments, I think they are mainly suggestions from the styling point of view :)

DelroyGayle committed:
Made changes to how the Grid displays images

Thank You Margarita

DelroyGayle committed:
Level 300

Create a separate branch from Master

DelroyGayle committed:
Level 350 changes

Added a new program showflicker.js/html/css
to test tvMaze's API

showflicker.js goes through 'every' available show in shows.js and displays all the episodes for each show for 20 seconds
When list exhausted, it will repeat from the beginning

DelroyGayle committed:
Level 400 done

DelroyGayle committed:
Repeatedly selecting the same episode

Had to make amendment to the Border Functionality when user selects the same episode more than once in row

DelroyGayle committed:
Minor changes

DelroyGayle committed:
Typo

DelroyGayle committed:
Scroll to the episode

DelroyGayle committed:
FIRST TRY

First attempt of Level 500 shows list

DelroyGayle committed:
Level 500: switch from Show Listing to Episodes Listing

DelroyGayle committed:
Many changes to implement Level 500

Changes to HTML/CSS & JS Code
At the stage to incorporate live search

DelroyGayle committed:
adding colour text

DelroyGayle committed:
Update script.js

Tested the regex

DelroyGayle committed:
FINALLY FIXED MANY MAJOR BUGS

DelroyGayle committed:
ALL BUGS FIXED!!

All the major bugs regarding filtering and displayed

DelroyGayle committed:
Changed the Shows Listing from FLEX to GRID

Applied CSS GRID

DelroyGayle committed:
REMOVED ALL CONSOLE.LOGS - LEVEL 500 DONE!

DelroyGayle committed:
Minor changes - Level 500 done

DelroyGayle committed:
pre-Mark 2

Made some cosmetic changes

DelroyGayle committed:
MARK 2

Better handling of shows with null images and null summary text

DelroyGayle committed:
Merge pull request #5 from DelroyGayle/Level500

Level500

DelroyGayle committed:
Updated README

DelroyGayle committed:
Update README.md - Added Live Link
